Use alternate mechanism to locate gcc include directory as recent
versions of gcc emit "Using built-in specs." rather than a path.

diff --exclude=debian -Naur perl-5.8.7.orig/utils/h2ph.PL perl-5.8.7/utils/h2ph.PL
--- perl-5.8.7.orig/utils/h2ph.PL	2005-04-05 07:47:17.000000000 +1000
+++ perl-5.8.7/utils/h2ph.PL	2005-11-13 11:06:37.000000000 +1100
@@ -734,10 +734,13 @@
 # non-GCC?) C compilers, but gcc uses an additional include directory.
 sub inc_dirs
 {
-    my $from_gcc    = `$Config{cc} -v 2>&1`;
-    $from_gcc       =~ s:^Reading specs from (.*?)/specs\b.*:$1/include:s;
-
-    length($from_gcc) ? ($from_gcc, $Config{usrinc}) : ($Config{usrinc});
+    my @inc = $Config{usrinc};
+    my $from_gcc = `$Config{cc} --print-libgcc-file-name 2>&1`;
+    if ($from_gcc =~ s!/libgcc\..*!/include!s and -d $from_gcc)
+    {
+	unshift @inc, $from_gcc;
+    }
+    @inc;
 }
 
 
